GrPothosBlock/pothos_pmt_helper.cc

```diff
@@ -300,12 +323,71 @@ Pothos::Object pmt_to_obj(const pmt::pmt_t &p)
 #include <Pothos/Plugin.hpp>
 #include <Pothos/Callable.hpp>
 
+// Use these functions to make direct conversions easier.
+
+template <typename T>
+static pmt::pmt_t convert_to_pmt(const T& input)
+{
+    return obj_to_pmt(Pothos::Object(input));
+}
+
+template <typename T>
+static T convert_from_pmt(const pmt::pmt_t& pmt)
+{
+    return pmt_to_obj(pmt).extract<T>();
+}
+
+template <typename T>
+static void register_converter_pair(const std::string& name)
+{
+    Pothos::PluginRegistry::add(
+        Poco::format("/object/convert/gr/%s_to_pmt", name),
+        Pothos::Callable(&convert_to_pmt<T>));
+    Pothos::PluginRegistry::add(
+        Poco::format("/object/convert/gr/%s_from_pmt", name),
+        Pothos::Callable(&convert_from_pmt<T>));
+}
+
 pothos_static_block(pothosObjectRegisterPMTSupport)
 {
+    // Note: don't add the reverse casts (pmt::to_bool, etc). This causes other primitive types to try
+    // converting through these PMT functions, resulting in exceptions.
```

Comment:
dang I guess I should do something about that. It must try to perform an intermediate conversion when there isnt a match. pmt->bool->desired type

Reply:
I'm also seeing numbers try to go through a string. It'd be great if there was a way to force a direct conversion or to prefer/force an intermediate type.

Reply:
I think i made that for numeric conversions and its kind of a pest. I almost never want to have it. Maybe I will make the types that should support intermediate conversions register in a different path.